### PR TITLE
[KOGITO-2859] make vet copy manifests to version

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-2859

This will be useful for the next release :)

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: hack/bump-version.sh [hunk 1]
```diff
@@ -26,23 +26,23 @@ if [ -z "$new_version" ]; then
 fi
 
 sed -i "s/$old_version/$new_version/g" cmd/kogito/version/version.go README.md version/version.go deploy/operator.yaml deploy/olm-catalog/kogito-operator/kogito-operator.package.yaml deploy/olm-catalog/kogito-operator/custom-subscription-example.yaml hack/go-build.sh hack/go-vet.sh .osdk-scorecard.yaml
-operator-sdk generate csv --apis-dir ./pkg/apis/apps/v1alpha1 --verbose --csv-version "$new_version" --from-version "$old_version" --update-crds --operator-name kogito-operator
 sed -i "s|operated-by: kogito-operator.*|operated-by: kogito-operator.${new_version}|g" deploy/olm-catalog/kogito-operator/manifests/kogito-operator.clusterserviceversion.yaml
 
+if [ "${old_version}" != "${new_version}" ]; then
```
Comment (radtriste) [inline hack/bump-version.sh:31]: to prevent any failure from the command if old_version == new_version

File: hack/bump-version.sh [hunk 2]
```diff
@@ -26,23 +26,23 @@ if [ -z "$new_version" ]; then
...
+    operator-sdk generate csv --apis-dir ./pkg/apis/app/v1alpha1 --verbose --csv-version "$new_version" --from-version "$old_version" --update-crds --operator-name kogito-operator
+fi
+make vet
```
Comment (radtriste) [inline hack/bump-version.sh:34]: This will generate all crds and csv + copy to correct versioned folder

File: hack/bump-version.sh [hunk 3]
```diff
@@ -26,23 +26,23 @@ if [ -z "$new_version" ]; then
...
-    image_version=`echo "${new_version}" | awk -F. '{print \$1"."\$2}'`
-    if [ "${no_release_branch}" = "true" ]; then
-        branch="master"
-    else
-        branch="${image_version}.x"
-    fi
-else
-    image_version="latest"
+
+image_version=`echo "${new_version}" | awk -F. '{print \$1"."\$2}'`
```
Comment (radtriste) [inline hack/bump-version.sh:39]: refactored it for more flexibility.
Default, we consider being on a release branch when bumping the version.
Rest of the parameters would define any change

File: deploy/olm-catalog/kogito-operator/0.13.0/kogito-operator.v0.13.0.clusterserviceversion.yaml [hunk 4]
```diff
@@ -101,9 +101,9 @@ metadata:
     capabilities: Basic Install
     categories: Integration & Delivery
     certified: "false"
-    containerImage: quay.io/kiegroup/kogito-cloud-operator:0.13.0
+    containerImage: quay.io/kiegroup/kogito-cloud-operator:0.10.0
```
Comment (sutaakar) [inline deploy/olm-catalog/kogito-operator/0.13.0/kogito-operator.v0.13.0.clusterserviceversion.yaml:104]: Is this intentional change?

Comment (ricardozanini) [reply deploy/olm-catalog/kogito-operator/0.13.0/kogito-operator.v0.13.0.clusterserviceversion.yaml:104]: I don't think so!

File: deploy/olm-catalog/kogito-operator/0.13.0/kogito-operator.v0.13.0.clusterserviceversion.yaml [hunk 5]
```diff
@@ -762,17 +762,14 @@ spec:
...
-    * Installs the [Kogito Jobs Service](https://docs.jboss.org/kogito/release/latest/html_single/#con-kogito-operator-with-jobs-service_kogito-deploying-on-openshift) to enable jobs scheduling.
+    * Creates two kinds of build configuration: Native or JVM. For more information refer to [Native X JVM Builds](https://github.com/kiegroup/kogito-cloud-operator#native-x-jvm-builds) on our docs
+    * Builds a Kogito Runtime Service from a remote git repository and creates a custom image for the given runtime: Quarkus or Springboot. [See more](https://github.com/kiegroup/kogito-cloud-operator#kogito-runtimes-service-deployment).
+    * Installs the [Kogito Data Index Service](https://github.com/kiegroup/kogito-runtimes/wiki/Data-Index-Service) to enable data index feature for Kogito Runtimes. [See more](https://github.com/kiegroup/kogito-cloud-operator#kogito-data-index-service-installation).
+    * Installs the [Kogito Jobs Service](https://github.com/kiegroup/kogito-runtimes/wiki/Jobs-Service) to enable jobs scheduling.
```
Comment (ricardozanini) [inline deploy/olm-catalog/kogito-operator/0.13.0/kogito-operator.v0.13.0.clusterserviceversion.yaml:770]: This is old, from where you took this CSV?

Comment (radtriste) [reply deploy/olm-catalog/kogito-operator/0.13.0/kogito-operator.v0.13.0.clusterserviceversion.yaml:770]: this was generated into the manifests folder by the `make vet` ... gonna check ...

File: deploy/olm-catalog/kogito-operator/0.13.0/kogito-operator.v0.13.0.clusterserviceversion.yaml (hunk 4 above)
Comment (radtriste) [reply deploy/olm-catalog/kogito-operator/0.13.0/kogito-operator.v0.13.0.clusterserviceversion.yaml:104]: https://github.com/kiegroup/kogito-cloud-operator/blob/f4c3e4f1b7c58df829b2ebee9b99aba911bfa02e/deploy/olm-catalog/kogito-operator/manifests/kogito-operator.clusterserviceversion.yaml#L104

File: deploy/olm-catalog/kogito-operator/0.13.0/kogito-operator.v0.13.0.clusterserviceversion.yaml (hunk 5 above)
Comment (radtriste) [reply deploy/olm-catalog/kogito-operator/0.13.0/kogito-operator.v0.13.0.clusterserviceversion.yaml:770]: https://github.com/kiegroup/kogito-cloud-operator/blob/f4c3e4f1b7c58df829b2ebee9b99aba911bfa02e/deploy/olm-catalog/kogito-operator/manifests/kogito-operator.clusterserviceversion.yaml#L767

Comment (radtriste) [reply deploy/olm-catalog/kogito-operator/0.13.0/kogito-operator.v0.13.0.clusterserviceversion.yaml:770]: I think the manifests csv file was never updated by hand ... only the versioned file.
We should enforce the manifests folder to be changed and then use `make vet`
Copying these 0.12 changes to manifests folder

File: deploy/olm-catalog/kogito-operator/0.13.0/kogito-operator.v0.13.0.clusterserviceversion.yaml (hunk 4 above)
Comment (radtriste) [reply deploy/olm-catalog/kogito-operator/0.13.0/kogito-operator.v0.13.0.clusterserviceversion.yaml:104]: updating the bump-version.sh script to update that one

**Comment by ricardozanini** [comment]
@radtriste unrelated, but the tag version should have the `v` on it. I saw that you created a milestone `0.14.0`, I'm supposing you used the same script to tag. The tag should be `v0.14.0`: https://godoc.org/golang.org/x/mod/semver

**Comment by radtriste** [comment]
@ricardozanini talking about the git tag right ? nothing related directly to this PR or ? gonna update the release pipeline with that

**Comment by ricardozanini** [comment]
> @ricardozanini talking about the git tag right ? nothing related directly to this PR or ? gonna update the release pipeline with that

Yes! Was not related to this PR, only to the pipeline script. :) I don't have access to the chat right now. hehe that's why I left this little note for you here. ❤️

**Comment by radtriste** [comment]
@ricardozanini https://github.com/radtriste/kogito-cloud-operator/releases for example :)

And I think this one can be merged :)

**Pull request merged**